Моделирование предпочтений

Для оценивания системы спроса на продовольственные товары
использована выборка RLMS, содержащая данные по социальноэкономическим характеристикам домохозяйств РФ. Данные RLMS
представлены в виде панели домохозяйств, расположенных в различных регионах РФ. При этом статистика собирается не по каждому отдельному человеку, а по всему домохозяйству, которое может
состоять из нескольких человек. Данные RLMS основаны на соцопросах, проводимых ежегодно среди выбранных домохозяйств. При
этом домохозяйство предоставляет данные по расходам и доходам
только за последний месяц. Таким образом, панель RLMS представляет месячные данные по расходам и доходам домохозяйств, отличающиеся субъективностью оценки, на каждый год рассматриваемого периода

Ссылка на данные RLMS-HSE

Согласно Условиям доступа к  данным RLMS-HSE, при использовании информации, полученной на основании RLMS-HSE, в публичных целях ссылка на источник должна быть следующей:

«Российский мониторинг экономического положения и здоровья населения НИУ ВШЭ (RLMS HSE)», проводимый Национальным исследовательским университетом  "Высшая школа экономики" и ООО «Демоскоп» при участии Центра народонаселения Университета Северной Каролины в Чапел Хилле и Института социологии Федерального научно-исследовательского социологического центра РАН. (Сайты обследования RLMS HSE: http://www.hse.ru/rlms и https://rlms-hse.cpc.unc.edu)».

In [1]:
import pandas as pd

In [6]:
df=pd.read_spss(r'C:\Users\79618\Desktop\Репрезентативная выборка 10.10.2021\29-я волна\ДОМОХОЗЯЙСТВА\r29h_os41.sav')

In [5]:
import pyreadstat

In [18]:
df.head()

YREDID_H    YID_H    XID_H    WID_H    VID_H    UID_H    TID_H    SID_H  \
0    1017.0  10003.0  10003.0  10003.0  10003.0  10003.0  10003.0  10003.0   
1    1018.0  10004.0  10004.0      NaN  10004.0  10004.0  10004.0  10004.0   
2    1020.0  10012.0  10012.0  10012.0  10012.0  10012.0  10012.0  10012.0   
3    1022.0  10014.0  10014.0  10014.0  10014.0  10014.0  10014.0  10014.0   
4    1023.0  10016.0  10016.0  10016.0  10016.0  10016.0  10016.0  10016.0   

     RID_H    QID_H  ...  yf14.12  yf14.13  yf14.14  yf14.15      yf14  \
0  10003.0  10003.0  ...      Нет      NaN      Нет      NaN   17376.0   
1  10004.0  10004.0  ...      Нет      NaN      Нет      NaN   37500.0   
2  10012.0  10012.0  ...      Нет      NaN      Нет      NaN   19700.0   
3  10014.0  10014.0  ...      Нет      NaN      Нет      NaN   15985.0   
4  10016.0  10016.0  ...      Нет      NaN      Нет      NaN  126000.0   

          yf16.1         yf16.2         yf16.3        yf16.4         yf16.5  
0   Нет, не было  НУЖДЫ НЕ БЫЛО   Нет, не было  Нет, не было  НУЖДЫ НЕ БЫЛО  
1   Нет, не было  НУЖДЫ НЕ БЫЛО  НУЖДЫ НЕ БЫЛО  Нет, не было       Да, было  
2  НУЖДЫ НЕ БЫЛО  НУЖДЫ НЕ БЫЛО  НУЖДЫ НЕ БЫЛО  Нет, не было       Да, было  
3   Нет, не было   Нет, не было   Нет, не было  Нет, не было   Нет, не было  
4   Нет, не было   Нет, не было   Нет, не было  Нет, не было  НУЖДЫ НЕ БЫЛО  

[5 rows x 1033 columns]

In [20]:
for col in df.columns:
    print(col)

YREDID_H
YID_H
XID_H
WID_H
VID_H
UID_H
TID_H
SID_H
RID_H
QID_H
PID_H
OID_H
NID_H
MID_H
LID_H
KID_H
JID_H
IID_H
HID_H
GID_H
FID_H
EID_H
DID_H
CID_H
BID_H
AID_H
y_origsm
y_hhwgt
region
psu
status
popul
site
ya3
ya3.3
Y_ROUND
YA4.1
YA4.2
YA5.1
YA5.2
YA8
YB1.O
YB1.N
y_nfm
YB1.1
yidind1
YB1.2.1
YB1.2.23
YB1.3
YB1.4
YB1.5
YB2.1
yidind2
YB2.2.1
YB2.2.23
YB2.3
YB2.4
YB2.5
YB2.9.1
YB3.1
yidind3
YB3.2.1
YB3.2.23
YB3.3
YB3.4
YB3.5
YB3.9.1
YB3.9.2
YB4.1
yidind4
YB4.2.1
YB4.2.23
YB4.3
YB4.4
YB4.5
YB4.9.1
YB4.9.2
YB4.9.3
YB5.1
yidind5
YB5.2.1
YB5.2.23
YB5.3
YB5.4
YB5.5
YB5.9.1
YB5.9.2
YB5.9.3
YB5.9.4
YB6.1
yidind6
YB6.2.1
YB6.2.23
YB6.3
YB6.4
YB6.5
YB6.9.1
YB6.9.2
YB6.9.3
YB6.9.4
YB6.9.5
YB7.1
yidind7
YB7.2.1
YB7.2.23
YB7.3
YB7.4
YB7.5
YB7.9.1
YB7.9.2
YB7.9.3
YB7.9.4
YB7.9.5
YB7.9.6
YB8.1
yidind8
YB8.2.1
YB8.2.23
YB8.3
YB8.4
YB8.5
YB8.9.1
YB8.9.2
YB8.9.3
YB8.9.4
YB8.9.5
YB8.9.6
YB8.9.7
YB9.1
yidind9
YB9.2.1
YB9.2.23
YB9.3
YB9.4
YB9.5
YB9.9.1
YB9.9.2
YB9.9.3
YB9.9.4
YB9.9.5
YB9.9.6
YB9.9.7
YB9.9.8
YB

y - код волны
y_nfm  - число членов семьи

In [ ]:
описательные статистики

In [39]:
df.loc[:,"y_nfm"].describe()

count    4831.000000
mean        2.592838
std         1.569647
min         1.000000
25%         1.000000
50%         2.000000
75%         3.000000
max        16.000000
Name: y_nfm, dtype: float64

In [ ]:
ax = df.plot.hist(column=["y_nfm"], by="gender", figsize=(10, 8))

таким образом среднее домохозяйство имеет 2.592838 членов

99999997 ЗАТРУДНЯЮСЬ ОТВЕТИТЬ
99999998 ОТКАЗ ОТ ОТВЕТА 
99999999 НЕТ ОТВЕТА 

Смотрю, чтобы не было NaN в данных по размеру домохозяйства

In [30]:
for i in range(0,len(df.loc[:,"y_nfm"])):
    if df.loc[:,"y_nfm"][i] in range(0,14):
        continue
    else:
        print(df.loc[:,"y_nfm"][i])

16.0
14.0


In [19]:
df.loc[:,"YREDID_H"]

0       1017.0
1       1018.0
2       1020.0
3       1022.0
4       1023.0
         ...  
4826      69.0
4827     206.0
4828    6157.0
4829    3229.0
4830    3230.0
Name: YREDID_H, Length: 4831, dtype: float64

Code-book RLMS-HSE
ID_W Волна (год проведения исследования) 
5 1994 год 
29 2020 год
! 1997 год пропущен
----
A3 НОМЕР СЕМЬИ 

A3.3 ХОТЯ БЫ ОДИН ЧЛЕН СЕМЬИ УЧАСТВОВАЛ В ИССЛЕДОВАНИИ